packages/core/src/contexts/reducers/store.ts

```diff
@@ -515,6 +519,48 @@ export const reducer = <S extends State>(state: S, action: ReducerAction<Dispatc
         preferences,
       }
     }
+    case PreferencesDispatchAction.SET_SELECTED_MEDIATOR: {
+      const selectedMediator = (action?.payload ?? []).pop() ?? ''
```

Comment:
Should the fallback here be the current mediator?
```suggestion
      const selectedMediator = (action?.payload ?? []).pop() ?? state.preferences.selectedMediator
```